ant/src/initialize.lua

```diff
@@ -0,0 +1,39 @@
+local utils = require(".utils")
+local initialize = {}
+
+function initialize.initialize(state)
+	local name, ticker, balances, controllers, records =
+		state.name, state.ticker, state.balances, state.controllers, state.records
+
+	assert(type(name) == "string", "name must be a string")
+	assert(type(ticker) == "string", "ticker must be a string")
+	assert(type(balances) == "table", "balances must be a table")
+	for k, v in pairs(balances) do
+		local idValidity, _ = pcall(utils.validateArweaveId, k)
+		assert(idValidity == false, "balances keys must be strings")
+		assert(type(v) == "number", "balances values must be numbers")
```

Comment:
nit: you can just do asserts directly, the validateArweaveId should throw the error and fail the `assert`

Reply:
Understood, the reasoning here was to specify WHICH field was invalid. We dont get stack traces in utils.reply.

Reply:
the first failed assert will be the `result`